operators/src/pro/cache/tile_cache.rs

```diff
@@ -307,38 +444,189 @@ impl TileCache {
-        let cache = caches.entry(key).or_default();
-        cache.active_queries.remove(&query_id);
+        let mut backend = self.backend.write().await;
+
+        let cache = backend.operator_caches.entry(key).or_default();
+        let entry = cache.landing_zone.remove(&query_id);
+
+        // update landing zone
+        if let Some(entry) = entry {
+            backend.landing_zone_byte_size_used -= entry.byte_size();
+        }
     }
 
     /// Finish the query and make the tiles available in the cache
     pub async fn finish_query(&self, key: CanonicOperatorName, query_id: QueryId) -> Result<()> {
-        let mut caches = self.operator_caches.write().await;
-        let cache = caches.entry(key).or_default();
+        // TODO: maybe check if this cache result is already in the cache or could displace another one
+
+        let mut backend = self.backend.write().await;
+
+        // this should always work, because the query was inserted at some point and then the cache entry was created
+        let cache = backend
+            .operator_caches
+            .get_mut(&key)
+            .ok_or(super::error::CacheError::QueryNotFoundInLandingZone)?;
+
         let active_query = cache
-            .active_queries
+            .landing_zone
             .remove(&query_id)
-            .ok_or(crate::error::Error::QueryProcessor)?; // TODO: better error
+            .ok_or(super::error::CacheError::QueryNotFoundInLandingZone)?;
+
+        let entry_size = active_query.byte_size(); // TODO is active query and cache entry the same size?
+
+        // move entry from landing zone into cache
+        let entry: CacheEntry = active_query.into();
+        let entry_id = CacheEntryId::new();
+        cache.entries.insert(entry_id, entry);
+
+        backend.lru.push(entry_id, key);
+
+        backend.landing_zone_byte_size_used -= entry_size;
+        // TODO: include the size of the entry in the lru and the operator_caches as well(?)
+        backend.cache_byte_size_used += entry_size;
+
+        // cache bound can be temporarily exceeded as the entry is moved form the landing zone into the cache
+        // but the total of cache + landing zone is still below the bound
+        // We now evict elements from the cache until bound is satisfied again
+        while backend.cache_byte_size_used + entry_size > backend.cache_byte_size_total {
+            // this should always work, because otherwise it would mean the cache is not empty but the lru is.
+            // the landing zone is smaller than the cache size and the entry must fit into the landing zone.
+            if let Some((id, op)) = backend.lru.pop_lru() {
+                if let Some(evict_cache) = backend.operator_caches.get_mut(&op) {
+                    if let Some(entry) = evict_cache.entries.remove(&id) {
+                        backend.cache_byte_size_used -= entry.byte_size();
+                    }
+                }
+            }
```

Comment:
This is super unreadable. Could you remove indentation layers, here?

Reply:
I removed two indentations layers: 2317aaf5501492506fcd12b5ba899a4520a3a9a5